4.7.1-windowsservercore-1709/build/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM microsoft/dotnet-framework-build:4.7.1-2017.11-windowsservercore-1709
+
+# Install web targets
+COPY ["Web", "/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/MSBuild/Microsoft/VisualStudio/v15.0/Web"]
+COPY ["WebApplications", "/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/MSBuild/Microsoft/VisualStudio/v15.0/WebApplications"]
+
+# Install node, bower, and git
+ENV NODE_VERSION 8.9.1
+RUN Invoke-WebRequest -UseBasicParsing https://nodejs.org/dist/v${env:NODE_VERSION}/node-v${env:NODE_VERSION}-win-x64.zip -outfile node.zip; \
+    Expand-Archive node.zip -DestinationPath nodejs-tmp; \
+    Move-Item nodejs-tmp/node-v${env:NODE_VERSION}-win-x64 ${env:ProgramFiles}/nodejs; \
+    Remove-Item -Force node.zip; \
+    Remove-Item -Force nodejs-tmp
+RUN setx /M PATH $(${Env:PATH} + \"${Env:ProgramFiles}/nodejs\")
+RUN & ${env:ProgramFiles}/nodejs/npm install -g gulp bower
+
+# Add ASP.NET runtime components
+RUN Add-WindowsFeature Web-Server; \
+    Add-WindowsFeature NET-Framework-45-ASPNET; \
+    Add-WindowsFeature Web-Asp-Net45; \
+    Remove-Item -Recurse C:\inetpub\wwwroot\*
+
+ADD ServiceMonitor.exe /
+
+# Download Roslyn nupkg and ngen the compiler binaries
+RUN Invoke-WebRequest https://api.nuget.org/packages/microsoft.net.compilers.2.3.1.nupkg -OutFile c:\microsoft.net.compilers.2.3.1.zip ; \
```

Comment:
Are these solely required for razor compilation? I was wondering if pre-compiled Roslyn is generally useful. If so, I'd suggest we put this command in the Docker image a layer below.

Reply:
I am of the opinion that the ASP.NET build image should contain the ASP.NET runtime (e.g. the same contents as the runtime image).  The build image isn't based on the runtime image as there is more to be shared by basing it on the Fx build image and then add the runtime versus the other way around.  That being said, Rich's question really applies to the runtime image.

@shirhatti - Can you explain the scenarios for these and if they should be separate into a separate image?

FYI this layer is 126MB.

Reply:
This was added as a performance optimization to improve time to first request.
If your project depends on `Microsoft.CodeDom.Providers.DotNetCompilerPlatform` (for Razor compilation) on the first request you pay a penalty of 1-3 seconds (on bare metal, probably more in a container) on the first request as csc.exe and VBCSCompiler.exe are NGENed.
You would ideally pre-compile your views but since views can be injected on the server at run time, it's a good idea to have this around.

Reply:
@Jinhuafei to keep me honest here